astroquery/hsa/core.py

```diff
@@ -0,0 +1,364 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import cgi
+import os
+import re
+import shutil
+from pathlib import Path
+
+from astroquery import log
+from astroquery.exceptions import LoginError, InputWarning
+from astroquery.query import BaseQuery
+from astroquery.utils.tap.core import Tap
+
+from . import conf
+
+__all__ = ['HSA', 'HSAClass']
+
+
+class HSAClass(BaseQuery):
+
+    data_url = conf.DATA_ACTION
+    metadata_url = conf.METADATA_ACTION
+    timeout = conf.TIMEOUT
+
+    def __init__(self, tap_handler=None):
+        super(HSAClass, self).__init__()
+        if tap_handler is None:
+            self._tap = Tap(url=self.metadata_url)
+        else:
+            self._tap = tap_handler
+
+    def download_data(self, *, retrieval_type=None,
```

Comment:
users may want to be able to set the download directory (e.g `download_dir` in esasky), but still let the filenames figured out automatically. Would be nice if this method and a few below would support it.

Reply:
Added `download_dir`